projects/components/src/viewer/code-viewer/code-viewer.component.ts

```diff
@@ -126,4 +154,26 @@ export class CodeViewerComponent implements OnChanges {
   public onSearch(searchText: string): void {
     this.searchText = searchText;
   }
+
+  private observeDomMutations(): void {
+    const codeViewerContentElement = this.element.nativeElement.querySelector('#code-viewer-content') as Node;
+    this.domMutationObserver.observe(codeViewerContentElement, {
+      subtree: true,
+      childList: true,
+      attributes: true,
+      attributeFilter: ['class']
+    });
+  }
+
+  private onDomMutation(mutations: MutationRecord[]): void {
+    const mutationType: MutationRecordType = mutations[0].type;
+
+    if (mutationType === 'childList') {
+      const markElement: HTMLElement = this.element.nativeElement.querySelector('mark');
+      markElement?.scrollIntoView();
+    } else if (mutationType === 'attributes') {
```

Comment:
Is this all to scroll into view on highlight? Doesn't' CDK give us some similar functionality without us building it ourselves? I guess a second question is - do we need to do this on the fly? This seems like it was an excuse to use dom mutation observer 😉

Reply:
heh.. Most of these dom elements are added dynamically after a search or highlight happens. In those cases, we don't get the reference to a new element unless they are actually added to the dom. Mutation observer is providing that callback through which we can deterministically call `scrollIntoView` on the first match

Reply:
ah search makes sense - with highlight, I was assuming that we can just onChange to know when to rescroll